web/auth/views.py

```diff
@@ -67,4 +140,7 @@ def login():
 @login_required
 def logout():
     logout_user()
-    return redirect("/")
+    idp_provider_cfg = get_idp_provider_cfg()
+    end_session_endpoint = idp_provider_cfg["end_session_endpoint"]
+    redirect_url = "{end_session_endpoint}?redirect_uri={redirect_uri}".format(end_session_endpoint=end_session_endpoint, redirect_uri=urllib.parse.quote(request.base_url))
+    return redirect(redirect_url)
```

Comment:
Code scanning complains about a reflected server-side XSS and URL redirection from remote source. This redirect accepts user input that isn't sanitized; could you take a look at re-writing this part which includes input validation and sanitation?

Reply:
@P-T-I, I have added a parse quote with safe param. Is it better now?

Reply:
Yes!

Reply:
Looks Ok!